pkg/plugins/runtime/k8s/controllers/configmap_controller.go

```diff
@@ -62,188 +62,141 @@ func (r *ConfigMapReconciler) Reconcile(req kube_ctrl.Request) (kube_ctrl.Result
-				Client:            mgr.GetClient(),
-				Log:               r.Log.WithName("dataplane-to-configmap-mapper"),
-				SystemNamespace:   r.SystemNamespace,
-				ResourceConverter: r.ResourceConverter,
-			},
-		}).
-		Watches(&kube_source.Kind{Type: &mesh_k8s.ZoneIngress{}}, &kube_handler.EnqueueRequestsFromMapFunc{
-			ToRequests: &ZoneIngressToMeshMapper{
-				Log:               r.Log.WithName("zone-ingress-to-configmap-mapper"),
-				SystemNamespace:   r.SystemNamespace,
-				ResourceConverter: r.ResourceConverter,
-			},
-		}).
-		Watches(&kube_source.Kind{Type: &mesh_k8s.VirtualOutbound{}}, &kube_handler.EnqueueRequestsFromMapFunc{
-			ToRequests: &VirtualOutboundToConfigMapsMapper{
-				Client:          mgr.GetClient(),
-				Log:             r.Log.WithName("virtualoutbound-to-configmap-mapper"),
-				SystemNamespace: r.SystemNamespace,
-			},
-		}).
-		Watches(&kube_source.Kind{Type: &mesh_k8s.ExternalService{}}, &kube_handler.EnqueueRequestsFromMapFunc{
-			ToRequests: &ExternalServiceToConfigMapsMapper{
-				Client:          mgr.GetClient(),
-				Log:             r.Log.WithName("external-service-to-configmap-mapperr"),
-				SystemNamespace: r.SystemNamespace,
-			},
-		}).
+		Watches(&kube_source.Kind{Type: &kube_core.Service{}}, kube_handler.EnqueueRequestsFromMapFunc(ServiceToConfigMapsMapper(mgr.GetClient(), r.Log, r.SystemNamespace))).
+		Watches(&kube_source.Kind{Type: &mesh_k8s.Dataplane{}}, kube_handler.EnqueueRequestsFromMapFunc(DataplaneToMeshMapper(r.Log, r.SystemNamespace, r.ResourceConverter))).
+		Watches(&kube_source.Kind{Type: &mesh_k8s.ZoneIngress{}}, kube_handler.EnqueueRequestsFromMapFunc(ZoneIngressToMeshMapper(r.Log, r.SystemNamespace, r.ResourceConverter))).
+		Watches(&kube_source.Kind{Type: &mesh_k8s.VirtualOutbound{}}, kube_handler.EnqueueRequestsFromMapFunc(VirtualOutboundToConfigMapsMapper(r.Log, r.SystemNamespace))).
+		Watches(&kube_source.Kind{Type: &mesh_k8s.ExternalService{}}, kube_handler.EnqueueRequestsFromMapFunc(ExternalServiceToConfigMapsMapper(r.Log, r.SystemNamespace))).
 		Complete(r)
 }
 
-type ServiceToConfigMapsMapper struct {
-	kube_client.Client
-	Log             logr.Logger
-	SystemNamespace string
-}
+func ServiceToConfigMapsMapper(client kube_client.Reader, l logr.Logger, ns string) kube_handler.MapFunc {
+	l = l.WithName("service-to-configmap-mapper")
+	return func(obj kube_client.Object) []kube_reconile.Request {
+		cause, ok := obj.(*kube_core.Service)
+		if !ok {
+			l.WithValues("dataplane", obj.GetName()).Error(errors.Errorf("wrong argument type: expected %T, got %T", cause, obj), "wrong argument type")
+			return nil
+		}
 
-func (m *ServiceToConfigMapsMapper) Map(obj kube_handler.MapObject) []kube_reconile.Request {
-	cause, ok := obj.Object.(*kube_core.Service)
-	if !ok {
-		m.Log.WithValues("dataplane", obj.Meta).Error(errors.Errorf("wrong argument type: expected %T, got %T", cause, obj.Object), "wrong argument type")
-		return nil
-	}
+		ctx := context.Background()
```

Comment:
we have access to context from `.Reconcile` function, is there any reason why we don't want to pass it and use it here?

Reply:
my bad, this function is not called from '.Reconcile'